src/validation.cpp

```diff
@@ -1414,7 +1414,10 @@ bool CheckBurnSpend(const CTransaction &tx, const CCoinsViewCache &inputs)
     Coin coin;
     for(size_t i = 0; i < tx.vin.size(); ++i)
     {
-        if (inputs.GetCoin(tx.vin[i].prevout, coin) && coin.out.scriptPubKey == Params().GetConsensus().burnAddress)
+        if (inputs.GetCoin(tx.vin[i].prevout, coin)
+        && (coin.out.scriptPubKey == Params().GetConsensus().burnAddress
+        || (coin.nHeight > Params().GetConsensus().FortCanningHeight
+        && coin.out.scriptPubKey == Params().GetConsensus().foundationShareScript)))
```

Comment:
I would not treat foundationShareScript as burn, we will manually send whatever is left to the burn address.